lib/deaf_grandma.rb

```diff
@@ -14,14 +14,29 @@ def run!
     loop do
       user_input = get_user_input
       p speak(user_input)
+      exit if @bye_counter == 3
     end
   end
 
 
   def speak(input)
+    if input.match(/\p{Lower}/)
```

Comment:
> \p{Lower}

Wow, I've never seen this before.  Standard approaches would be 
```ruby
  input == input.downcase
```
or 
```ruby
  input.match /\A[a-z]*\z/
```
(or something like that, I'm not actually testing these things out.)

But if you can point me to do the relevant docs, I'm always happy to pick up another option.

Reply:
I was using an option that was gentle on non-alphabet characters but I achieved the same thing by using upcase as well. \p is a regex way of matching Unicode categories but it's not fully supported across languages (https://www.regular-expressions.info/unicode.html#category). It's supported in Ruby since 1.9

Reply:
🆒